statsd_client.ml

```diff
@@ -4,6 +4,21 @@ let port = ref None
 let log_debug = ref (fun s -> ())
 let log_error = ref (fun s -> Printf.eprintf "[error] %s\n%!" s)
 
+let fmt_guage = Printf.sprintf "%d|g"
+let fmt_time = Printf.sprintf "%d|ms"
+let fmt_counting = Printf.sprintf "%d|c"
+let sec_to_ms time = int_of_float (time *. 1000.)
+
+let fmt_statsd_payload ~sample_rate (metric_name, value_type) =
+  (* Map sample_rate to the string statsd expects *)
+  let sample_rate =
+    if sample_rate >= 1.0 then
+      ""  (* not sampling *)
```

Comment:
This is confusing. Why does >= 1 mean we don't send anything?

Reply:
NOTE: This logic is copied from before with no changes. 

Not sampling here just means that we are sending all data along. I.e. we have a sampling of 100% or 1.0. This logic seems faulty to me since sample-rate is a float and a negative sampling doesn't make sense. Nor does anything greater than 1.0. However, the error handling in this library is pretty poor, imo so its not accounted for. The way that dogstatsd handles this is as follows: https://github.com/DataDog/datadogpy/blob/eff9a2a03f92979fa17011df5b63a926a19802e2/datadog/dogstatsd/base.py#L279